### PR TITLE
[Fix] `no-unused-prop-types`: returns no errors when using the spread operator in JSX element

### LINKED ISSUE #3566: [Bug]: react/no-unused-prop-types returns no errors when using the spread operator in JSX element
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### Description Overview

When using the spread operator within a JSX component, the rule `react/no-unused-prop-types` no longer detects any errors.

## Demo Code

`package.json`
```json
{
  "scripts": {
    "lint": "eslint Demo.tsx"
  },
  "private": true,
  "devDependencies": {
    "@types/react": "^18.2.0",
    "@typescript-eslint/parser": "^5.59.1",
    "eslint": "^8.39.0",
    "eslint-plugin-react": "^7.32.2",
    "typescript": "^5.0.4"
  },
  "dependencies": {
    "react": "^18.2.0"
  }
}
```

`.eslintrc.js`
```js
module.exports = {
  parser: "@typescript-eslint/parser",
  extends: ["plugin:react/recommended"],
  parserOptions: {
    ecmaVersion: 2018,
    sourceType: "module",
    ecmaFeatures: {
      jsx: true
    }
  },
  settings: {
    react: {
      version: "detect"
    }
  },
  rules: {
    "react/no-unused-prop-types": 2
  }
};
```

`Demo.tsx`
```ts
import React from "react";

type Props = {
  foo: string;
  bar: string; // <-- unused prop
};

const Demo: React.FC<Props> = ({ foo }) => {
  return <div {...{}}>{foo}</div>;
};

export default Demo;

```

### With spread operator

No error is detected when a prop type is unused within the component a spread operator is used in an element for attributes.

CleanShot 2023-04-28 at 10 48 47@2x

### Without spread operator

When removing the spread operator, the unused prop is detected
CleanShot 2023-04-27 at 14 10 33@2x

## Steps to reproduce

`yarn lint`

### Expected Behavior

`bar` should be marked as a linting error
CleanShot 2023-04-27 at 14 08 06@2x

### eslint-plugin-react version

v7.32.2

### eslint version

v8.39.0

### node version

v18.11.0

### PR DESCRIPTION
**Pull request by akulsr0**
Fixes #3566

### CODE REVIEW

File: lib/util/usedPropTypes.js [hunk 1]
```diff
@@ -534,7 +534,7 @@ module.exports = function usedPropTypesInstructions(context, components, utils)
     JSXSpreadAttribute(node) {
       const component = components.get(utils.getParentComponent());
       components.set(component ? component.node : node, {
-        ignoreUnusedPropTypesValidation: true,
+        ignoreUnusedPropTypesValidation: !(node.argument.type === 'ObjectExpression'),
```
Comment (ljharb) [inline lib/util/usedPropTypes.js:537]:
```suggestion
        ignoreUnusedPropTypesValidation: node.argument.type !== 'ObjectExpression',
```

File: tests/lib/rules/no-unused-prop-types.js [hunk 2]
```diff
@@ -6689,6 +6689,27 @@ ruleTester.run('no-unused-prop-types', rule, {
...
+      type props = {
+        foo: string;
+        bar: string;
+      };
+
+      const Demo: React.FC<props> = ({ foo }) => {
+        return <div {...{}}>{foo}</div>;
+      };
+
+      export default Demo;
```
Comment (ljharb) [inline tests/lib/rules/no-unused-prop-types.js:6707]:
```suggestion
    {
      code: `
        import React from "react";

        type props = {
          foo: string;
          bar: string;
        };

        const Demo: React.FC<props> = ({ foo }) => {
          return <div {...{}}>{foo}</div>;
        };

        export default Demo;
```

**Review by ljharb — APPROVED** [review]
LGTM assuming tests pass!

**Comment by ljharb** [comment]
Looks like it's failing in eslint < 8 - perhaps the node type is different there?

**Comment by akulsr0** [comment]
I have fixed failing tests, there are still some tests failing probably an nvm issue since nvm install failed.

**Comment by akulsr0** [comment]
> I have fixed failing tests, there are still some tests failing probably an nvm issue since nvm install failed.

@ljharb Should we try rerunning the checks once?

**Comment by ljharb** [comment]
Sure, done.

**Pull request merged**